examples/lstm_text_generation.py

```diff
@@ -0,0 +1,230 @@
+
+The loss function compares (via categorical crossentropy) the prediction
+with the output/target.
+
+Also included is a function to generate text using the RNN given the first 
+character.  
+
+About 20 or so epochs are necessary to generate text that "makes sense".
+
+Written by @keskarnitish
+Pre-processing of text uses snippets of Karpathy's code (BSD License)
+'''
+
+from __future__ import print_function
+
+
+import numpy as np
+import theano
+import theano.tensor as T
+import lasagne
+
+try:
+    fname = 'input.txt' 
+    in_text = open(fname, 'r').read() # should be simple plain text file
+    in_text = in_text.decode("utf-8-sig").encode("utf-8")
+except IOError as e:
+    print("I/O error({0}): {1}".format(e.errno, e.strerror))
+    print("{0} does not exist. Please provide a valid txt file as input.".format(fname))
+    print("A sample txt file can be downloaded from https://s3.amazonaws.com/text-datasets/nietzsche.txt")
+    raise IOError('File Not Found')
+
+generation_phrase = "The quick brown fox jumps" #This phrase will be used as seed to generate text.
+
+#This snippet loads the text file and creates dictionaries to 
+#encode characters into a vector-space representation and vice-versa. 
+chars = list(set(in_text))
+data_size, vocab_size = len(in_text), len(chars)
+char_to_ix = { ch:i for i,ch in enumerate(chars) }
+ix_to_char = { i:ch for i,ch in enumerate(chars) }
+
+#Lasagne Seed for Reproducibility
+lasagne.random.set_rng(np.random.RandomState(1))
+
+# Sequence Length
+SEQ_LENGTH = 20
+
+# Number of units in the two hidden (LSTM) layers
+N_HIDDEN = 512
+
+# Optimization learning rate
+LEARNING_RATE = .01
+
+# All gradients above this will be clipped
+GRAD_CLIP = 100
+
+# How often should we check the output?
+PRINT_FREQ = 1000
+
+# Number of epochs to train the net
+NUM_EPOCHS = 50
+
+# Batch Size
+BATCH_SIZE = 128
+
+
+def gen_data(p, batch_size = BATCH_SIZE, data=in_text):
+    '''
+    This function produces a semi-redundant batch of training samples from the location 'p' in the provided string (data).
+    For instance, assuming SEQ_LENGTH = 5 and p=0, the function would create batches of 
+    5 characters of the string (starting from the 0th character and stepping by 1 for each semi-redundant batch)
+    as the input and the next character as the target.
+    To make this clear, let us look at a concrete example. Assume that SEQ_LENGTH = 5, p = 0 and BATCH_SIZE = 2
+    If the input string was "The quick brown fox jumps over the lazy dog.",
+    For the first batch,
+    x (the inputs to the neural network) would correspond to the encoding of 'T','h','e',' ','q'
+    y (the targets of the neural network) would be the encoding of 'u'
+    For the second batch,
+    x (the inputs to the neural network) would correspond to the encoding of 'h','e',' ','q', 'u'
+    y (the targets of the neural network) would be the encoding of 'i'
+    The batches are then stacked (into a three-dimensional tensor) and returned. 
```

Comment:
And `The data points are then stacked`. What you return is a single batch, i.e., a tensor of multiple data points.